src/sap/fhir/model/r4/FHIRUtils.js

```diff
@@ -605,6 +606,52 @@ sap.ui.define([
 		return oBinding && oBinding.getMetadata().getName() === "sap.fhir.model.r4.FHIRPropertyBinding";
 	};
 
+	/**
+	 * Generate FullUrl based on the type and id
+	 *
+	 * @param {sap.fhir.model.r4.lib.FHIRBundleEntryFullUrlType} sFullUrlType fullUrlType
+	 * @param {string} sResourceServerPath The original resource path e.g. /Patient/123
+	 * @param {string} sResourceId The id of the the FHIR resource e.g. 123
+	 * @param {string} sServiceUrl The root URL of the FHIR server to request data e.g. http://example.com/fhir
+	 * @returns {string} sFullUrl
+	 * @protected
+	 * @since 1.0.0
+	 */
+	FHIRUtils.generateFullUrl = function(sFullUrlType,sResourceServerPath,sResourceId,sServiceUrl) {
+		var sFullUrl;
+		if (sFullUrlType){
+			switch (sFullUrlType) {
+				case FHIRBundleEntryFullUrlType.uuid:
+					if (sResourceId && this.resourceIdIsOfTypeUUID(sResourceId)){
+						sFullUrl = "urn:" + FHIRBundleEntryFullUrlType.uuid + ":" + sResourceId;
+					} else {
+						sFullUrl = "urn:" + FHIRBundleEntryFullUrlType.uuid + ":" + this.uuidv4();
+					}
+					break;
+				case FHIRBundleEntryFullUrlType.absolute:
+					sFullUrl = sServiceUrl + sResourceServerPath;
+					break;
+				default:
+					sFullUrl = sResourceServerPath;
+					break;
+			}
+		}
+		return sFullUrl;
+
+	};
+
+	/**
+	 * Determines if the given resourceId is of type uuid v4
+	 *
+	 * @param {string} sResourceId The id of the the FHIR resource e.g. 123
+	 * @returns {boolean} True, if the expression matches the regex.
+	 * @protected
+	 * @since 1.0.0
+	 */
+	FHIRUtils.resourceIdIsOfTypeUUID = function(sResourceId){
```

Comment:
rename method to `isResourceIdOfTypeURI(sResourceId, sURIType)`. With that the method can be easily expanded in feature to cover all type checks

Reply:
not needed anymore since validateValue From ui5 model is used